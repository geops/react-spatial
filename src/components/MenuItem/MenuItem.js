import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  classNameTitle: PropTypes.string,
};

const defaultProps = {
  className: 'tm-menu-item',
  classNameTitle: 'tm-menu-item-title',
};

const MenuItem = ({ children, title, className, classNameTitle }) => (
  <>
    <div className={classNameTitle}>{title}</div>
    <div className={className}>{children}</div>
  </>
);

MenuItem.propTypes = propTypes;
MenuItem.defaultProps = defaultProps;

export default MenuItem;
