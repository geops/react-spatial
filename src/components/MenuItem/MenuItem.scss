.tm-menu .tm-menu-item {
  padding: 0 20px 20px 20px;
}
