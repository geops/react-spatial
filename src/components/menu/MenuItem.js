import React from 'react';
import PropTypes from 'prop-types';

import './MenuItem.scss';

const propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

const MenuItem = ({ children, title }) => (
  <>
    <div className="tm-menu-item-title">{title}</div>
    <div className="tm-menu-item">{children}</div>
  </>
);

MenuItem.propTypes = propTypes;

export default MenuItem;
