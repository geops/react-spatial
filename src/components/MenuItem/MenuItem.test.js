import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MenuItem from './MenuItem';

configure({ adapter: new Adapter() });

describe('MenuItem', () => {
  test('should match snapshot.', () => {
    const component = renderer.create(
      <MenuItem title="Test2">
        <div>test</div>
      </MenuItem>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
