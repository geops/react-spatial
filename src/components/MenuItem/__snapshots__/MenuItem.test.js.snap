// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should match snapshot. 1`] = `
Array [
  <div
    className="tm-menu-item-title"
  >
    Test2
  </div>,
  <div
    className="tm-menu-item"
  >
    <div>
      test
    </div>
  </div>,
]
`;
